app/socket-apis/__tests__/subscribe-deliberation.js

```diff
@@ -0,0 +1,98 @@
+// https://github.com/EnCiv/civil-pursuit/issues/196
+
+/**
+ * @jest-environment jsdom
+ */
+
+import { Mongo } from '@enciv/mongo-collections'
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import { MongoClient, ObjectId } from 'mongodb'
+import { Iota, serverEvents } from 'civil-server'
+import jestSocketApiSetup from '../../jest-socket-api-setup'
+import socketApiSubscribe, { subscribeEventName } from '../socket-api-subscribe'
+import { Discussions, initDiscussion, insertStatementId } from '../../dturn/dturn'
+
+const handle = 'subscribe-deliberation'
+const socketApiUnderTest = subscribeDeliberation
+
+const userId = '6667d5a33da5d19ddc304a6b'
+const synuser = { synuser: { id: userId } }
+
+const discussionId = '66a174b0c3f2051ad387d2a6'
+
+let MemoryServer
+
+import subscribeDeliberation from '../subscribe-deliberation'
+
+beforeEach(async () => {
+  jest.spyOn(console, 'error').mockImplementation(() => {})
+})
+
+afterEach(async () => {
+  console.error.mockRestore()
+})
+
+beforeAll(async () => {
+  MemoryServer = await MongoMemoryServer.create()
+  const uri = MemoryServer.getUri()
+  await Mongo.connect(uri)
+
+  await jestSocketApiSetup(userId, [[handle, socketApiUnderTest]])
+})
+
+afterAll(async () => {
+  Mongo.disconnect()
+  MemoryServer.stop()
+  window.socket.close()
+})
+
+// Tests
+test('Fail if user is not logged in.', async () => {
+  await subscribeDeliberation.call({}, discussionId)
+
+  expect(console.error.mock.calls[0][0]).toMatch(/user is not logged in/)
+})
+
+test('Fail if deliberation ID not provided.', async () => {
+  await subscribeDeliberation.call(synuser, null)
+
+  expect(console.error.mock.calls[0][0]).toMatch(/DeliberationId was not provided/)
+})
+
+test('Succeed if deliberation exists.', done => {
+  const anIota = {
+    _id: new ObjectId(discussionId),
+    path: '/deliberation1',
+    subject: 'Deliberation',
+    description: 'A descriptive discussion.',
+    webComponent: {
+      dturn: {
+        max_rounds: 3,
+      },
+    },
+  }
+
+  Iota.create(anIota)
+    .then(() => {
+      function requestHandler(data) {
+        console.log(data['participants'])
+        console.log('Request was called')
+        done()
+      }
+
+      function updateHandler(data) {
+        // Remove from memory if no subscribers remain
```

Comment:
In this jest test, the update handler should test that it received {participants, lastRound} correctly. 
Also, the update handler won't get called until after the request handler has been called and then, only after there has been some update that has been broadcast.  

This test case is setting up the dturn and checking that request handler is called.

Another test will have to submit data to dturn (like insert a statement) and then see that upddatehandler has been called.

The tricky part is that you have to pass updateHandler in the first test, and then check to see that it's been called in the second test.

Reply:
Hm, I found a few more issues and verified updateUInfo was properly called after the fixes - `server.to(eventName).emit(message)` and the next lines are run when I insert a statement with new changes, but updateHandler still isn't called - trying to figure out why.